edk2toollib/uefi/telem/parser_main.py

```diff
@@ -0,0 +1,660 @@
+        # This could be because there was an error parsing a section, or the
+        # section count is incorrect
+        if neq(self.Header.SectionCount,self.SectionHeaders.count):
+            print("Section Count of CPER header \n")
+
+        # Print each section header followed by the correlated section
+        for s in self.SectionHeaders:
+            print(concat("Section ",str(counter)))
+            print(s.PrettyPrint())
+            print(self.ParseSectionData(s))
+            counter += 1
+
+# CPER: Common Platform Error Record
+
+# Structure of a CPER Header:
+# Signature           (0   byte offset, 4  byte length) : CPER Signature
+# Revision            (4   byte offset, 2  byte length) : The Revision of CPER
+# Signature End       (6   byte offset, 4  byte length) : Must always be 0xffffffff
+# Section Count       (10  byte offset, 4  byte length) : The Number of Sections of CPER
+# Error Severity      (12  byte offset, 4  byte length) : The Severity of Error of CPER
+# Validation Bits     (16  byte offset, 4  byte length) : Identify Valid IDs of the CPER
+# Record Length       (20  byte offset, 4  byte length) : The size(in bytes) of the ENTIRE Error Record
+# Timestamp           (24  byte offset, 8  byte length) : The Time at which the Error occured
+# Platform ID         (32  byte offset, 16 byte length) : The Platform GUID (Typically, the Platform SMBIOS UUID)
+# Partition ID        (48  byte offset, 16 byte length) : The Software Partition (if applicable)
+# Creator ID          (64  byte offset, 16 byte length) : The GUID of the Error "Creator"
+# Notification Type   (80  byte offset, 16 byte length) : A pre-assigned GUID associated with the event (ex.Boot)
+# Record ID           (96  byte offset, 8  byte length) : When combined with the Creator ID, identifies the Error Record
+# Flags               (104 byte offset, 4  byte length) : A specific "flag" for the Error Record. Flags are pre-defined values which provide more information on the Error
+# Persistence Info    (108 byte offset, 8  byte length) : Produced and consumed by the creator of the Error Record.There are no guidelines for these bytes.
+# Reserved            (116 byte offset, 12 byte length) : Must be zero
+class CPER_HEADER(object):
+
+    STRUCT_FORMAT = "=IHIHIIIQ16s16s16s16sQIQ12s"
+
+    def __init__(self, input):
+        
+        self.ValidBitsList = [False,False,False] # Go to self.ValidBitsParse to see description of the field
+
+        (self.SignatureStart,
+         self.Revision,
+         self.SignatureEnd,
+         self.SectionCount,
+         self.ErrorSeverity,
+         self.ValidationBits,
+         self.RecordLength,
+         self.Timestamp,
+         self.PlatformID,
+         self.PartitionID,
+         self.CreatorID,
+         self.NotificationType,
+         self.RecordID,
+         self.Flags,
+         self.PersistenceInfo,
+         self.Reserved) = struct.unpack_from(self.STRUCT_FORMAT, input)
```

Comment:
what happens if this struct unpacking fails. Do we throw an exception? Does it get caught somewhere?